src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.wasm.cs

```diff
@@ -74,38 +71,51 @@ public static void SetBorder(UIElement element, Thickness thickness, Brush brush
 			else
 			{
 				var borderWidth = $"{thickness.Top.ToStringInvariant()}px {thickness.Right.ToStringInvariant()}px {thickness.Bottom.ToStringInvariant()}px {thickness.Left.ToStringInvariant()}px";
+
 				switch (brush)
 				{
 					case SolidColorBrush solidColorBrush:
-						var borderColor = solidColorBrush.ColorWithOpacity;
-						element.SetStyle(
-							("border", ""),
-							("border-style", "solid"),
-							("border-color", borderColor.ToHexString()),
-							("border-width", borderWidth));
+						ApplySolidColor(element, solidColorBrush.ColorWithOpacity, borderWidth);
+						break;
+					case LinearGradientBrush linearGradientBrush:
+						if (cornerRadius == CornerRadius.None ||
+							!BorderGradientBrushHelper.CanApplySolidColorRendering(linearGradientBrush))
+						{
+							ApplyGradient(element, linearGradientBrush, borderWidth);
+						}
+						else
+						{
+							var majorStop = BorderGradientBrushHelper.GetMajorStop(linearGradientBrush);
+							var borderColor = Color.FromArgb((byte)(linearGradientBrush.Opacity * majorStop.Color.A), majorStop.Color.R, majorStop.Color.G, majorStop.Color.B);
+
+							ApplySolidColor(element, borderColor, borderWidth);
+						}
 						break;
 					case GradientBrush gradientBrush:
-						var border = gradientBrush.ToCssString(element.RenderSize); // TODO: Reevaluate when size is changing
-						element.SetStyle(
-							("border-style", "solid"),
-							("border-color", ""),
-							("border-image", border),
-							("border-width", borderWidth),
-							("border-image-slice", "1"));
+						ApplyGradient(element, gradientBrush, borderWidth);
 						break;
 					case AcrylicBrush acrylicBrush:
 						var acrylicFallbackColor = acrylicBrush.FallbackColorWithOpacity;
-						element.SetStyle(
-							("border", ""),
-							("border-style", "solid"),
-							("border-color", acrylicFallbackColor.ToHexString()),
-							("border-width", borderWidth));
+						ApplySolidColor(element, acrylicFallbackColor, borderWidth);
```

Comment:
You should use, instead, the `Brush.TryGetColorWithOpacity(brush, out var color)`.